stroom-connectors/src/main/java/stroom/connectors/ConnectorProperties.java

```diff
@@ -0,0 +1,55 @@
+package stroom.connectors;
+
+import java.util.Properties;
+
+/**
+ * We need to expose properties to the connector implementations, but without them having to wire in our relatively heavy
+ * property service. This interface will be used to hide those details, and we will provide implementations
+ * that expect prefixed properties to be available in StroomPropertyService.
+ */
+public interface ConnectorProperties {
+    /**
+     * Extract the String value of a property for a given key.
+     *
+     * @param key The property key
+     * @return The value of the property as a string.
+     */
+    String getProperty(String key);
```

Comment:
I would be inclined to use Optional<...> for the return type of the 3 get methods.  Big fan of optional.  Nice and explicit about what you are getting back.